### PR TITLE
feat(middleware.cors): add CORS middleware

### PR DESCRIPTION
**Pull request by ziadsawalha**
This middleware simplifies adding support for CORS
header responses in bottle.

### CODE REVIEW

**Comment by samstav** [comment]
LGTM

**Comment by cbfx** [comment]
Reviewing now.

File: simpl/middleware/cors.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    webob = None
+
+LOG = logging.getLogger(__name__)
+
+
+class CORSMiddleware(object):  # pylint: disable=R0903
+
+    """Responds to CORS requests."""
+
+    default_methods = ('GET', 'OPTIONS', 'POST', 'PUT', 'HEAD')
```
Comment (cbfx) [inline simpl/middleware/cors.py]: I haven't gone all the way through this but shouldn't 'DELETE' be here too?

Comment (ziadsawalha) [inline simpl/middleware/cors.py]: Yes. Fixed. Thanks!

**Comment by Gifflen** [comment]
This is a great addition. Thanks @ziadsawalha cors has always been a pain. :+1:

**Comment by cbfx** [comment]
:shipit:

**Comment by cbfx** [comment]
@ziadsawalha @samstav @jmontemayor when will this be available in juno?

**Pull request merged**